src/sentry/integrations/opsgenie/integration.py

```diff
@@ -51,8 +62,92 @@
 )
 
 
+class OpsgenieFinishedView(PipelineView):
+    def dispatch(self, request: Request, pipeline: Pipeline) -> Response:
+        response = pipeline.finish_pipeline()
+
+        integration = getattr(pipeline, "integration", None)
+        if not integration:
+            return response
+
+        messages.add_message(request, messages.SUCCESS, "Opsgenie integration installed.")
+
+        return self.redirect(
+            absolute_uri(
+                f"/settings/{pipeline.organization.slug}/integrations/opsgenie/{integration.id}/"
+            )
+        )
+
+
+class InstallationForm(forms.Form):
+    base_url = forms.CharField(
+        label=_("Base URL"),
+        help_text=_("Either https://api.opsgenie.com/ or https://api.eu.opsgenie.com/"),
+        widget=forms.TextInput(attrs={"placeholder": "https://api.opsgenie.com/"}),
+    )
+    api_key = forms.CharField(
+        label=("Opsgenie API Key"),
+        widget=forms.TextInput(
+            attrs={
+                "placeholder": _("5832fc6e14300a0d962240a8144466eef4ee93ef0d218477e55f11cf12fc3737")
+            }
+        ),
+    )
+
+
+class InstallationConfigView(PipelineView):
+    def dispatch(self, request: Request, pipeline) -> HttpResponse:
+        if "goback" in request.GET:
+            pipeline.state.step_index = 0
+            return pipeline.current_step()
+        if request.method == "POST":
+            form = InstallationForm(request.POST)
+            if form.is_valid():
+                form_data = form.cleaned_data
+
+                pipeline.bind_state("installation_data", form_data)
+
+                return pipeline.next_step()
+        else:
+            form = InstallationForm()
+
+        return render_to_response(
+            template="sentry/integrations/opsgenie-config.html",
+            context={"form": form},
+            request=request,
+        )
+
+
+class InstallationGuideView(PipelineView):
+    def dispatch(self, request: Request, pipeline) -> HttpResponse:
+        if "completed_installation_guide" in request.GET:
+            return pipeline.next_step()
+        return render_to_response(
+            template="sentry/integrations/opsgenie-config.html",
+            context={
+                "next_url": f'{absolute_uri("/extensions/opsgenie/setup/")}?completed_installation_guide',
+                "setup_values": [
+                    {"label": "Name", "value": "Sentry"},
+                    {
+                        "label": "Access rights",
+                        "value": "Read, Create and update, Delete, Configuration access",
+                    },
+                ],
+            },
+            request=request,
+        )
+
+
 class OpsgenieIntegration(IntegrationInstallation):
-    pass
+    def update_organization_config(self, data: MutableMapping[str, Any]) -> None:
+        """
```

Comment:
What does this docstring mean? I assume its like an example config, but it could be confusing for people to come across later on

Reply:
I can move it to notion